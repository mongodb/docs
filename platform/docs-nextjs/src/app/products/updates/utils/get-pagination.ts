/**
 *
 * @param array - flatten array
 * @param chunkSize - max size of each array
 * @returns [[{}, {}], [{}]]
 * @description Takes an flatten array and creates one level nested array of objects based on the chunkSize. Used for pagination data.
 */
export const getPagination = <T>(array: T[], chunkSize: number): T[][] => {
  const chunks: T[][] = [];
  if (!array.length) {
    return [[]]; // default to one empty nested array
  }
  for (let i = 0; i < array.length; i += chunkSize) {
    chunks.push(array.slice(i, i + chunkSize));
  }
  return chunks;
};
