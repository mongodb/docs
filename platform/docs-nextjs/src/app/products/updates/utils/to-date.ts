export const normalizeDate = (date: string) => {
  // Check if the date string has the correct format 'YYYY-MM-DD'
  const dateToNormalize = /^(\d{4})-(\d{2})-(\d{2})$/.exec(date)?.slice(1); // excludes the full date in the first index i.e. "2025-05-27"
  if (dateToNormalize) {
    // This is needed because, JavaScript’s Date constructor uses zero-based months (January = 0 … December = 11).
    return new Date(Number(dateToNormalize[0]), Number(dateToNormalize[1]) - 1, Number(dateToNormalize[2]));
  }
  return new Date(date);
};
