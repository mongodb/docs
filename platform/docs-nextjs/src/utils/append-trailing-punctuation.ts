import { ASTNode, ReferenceNode, RefRoleNode } from "@/types/ast";
import { isReferenceNode, isRefRoleNode, isTextNode } from "@/types/ast-utils";

// Append dangling punctuation to preceding Link's text
export const appendTrailingPunctuation = (nodes: ASTNode[]) => {
  const truncatedNodes = [];

  for (let i = 0; i < nodes.length; i++) {
    const currNode = nodes[i];
    const nextNode = nodes[i + 1];
    const hasDanglingSibling =
      isTextNode(nextNode) &&
      nextNode.value.length === 1 &&
      !nextNode.value.match(/\s/);
    // TODO: if issues arise, instead of matching for non-whitespace, we might consider checking FOR certain singular punctuation
    // ie: regex match for  .?!,:;)*"']}`

    if ((isReferenceNode(currNode) || isRefRoleNode(currNode)) && hasDanglingSibling) {
      const copyOfNodeWithDeepCopiedChildren = deepCopyNodesChildren(currNode);
      copyOfNodeWithDeepCopiedChildren.children.push(nextNode);
      truncatedNodes.push(copyOfNodeWithDeepCopiedChildren);
      i++;
      continue;
    }
    truncatedNodes.push(currNode);
  }
  return truncatedNodes;
};

// Make a copy of node with a deep copy of the new child node
function deepCopyNodesChildren(refNode: ReferenceNode | RefRoleNode) {
  if (!refNode.children || !refNode.children.length) return { ...refNode };
  const copyOfChildren = [...refNode.children];

  const copyOfNode = {
    ...refNode,
    children: copyOfChildren,
  };
  return copyOfNode;
}
