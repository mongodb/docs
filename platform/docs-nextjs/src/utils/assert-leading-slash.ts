const assertLeadingSlash = (url: string) => {
  if (url && url.match(/^\//)) {
    return url;
  }
  return `/${url}`;
};

export { assertLeadingSlash };
