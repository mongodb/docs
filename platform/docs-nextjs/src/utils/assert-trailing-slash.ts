export const assertTrailingSlash = (url: string) => {
  if (url && url.match(/\/$/)) {
    return url;
  }
  return `${url}/`;
};
