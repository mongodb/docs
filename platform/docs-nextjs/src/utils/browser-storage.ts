import { isBrowser } from './is-browser';

const isValidStorage = isBrowser;

export const setLocalValue = (key: string, value: unknown) => {
  try {
    if (isValidStorage) {
      const prevState = JSON.parse(window.localStorage.getItem('mongodb-docs') || '{}');
      localStorage.setItem('mongodb-docs', JSON.stringify({ ...prevState, [key]: value }));
    }
  } catch {
    console.error('Error setting localStorage value');
  }
};

export const getLocalValue = (key: string) => {
  try {
    if (isValidStorage && JSON.parse(window.localStorage.getItem('mongodb-docs') || '{}')) {
      const docsObj = JSON.parse(window.localStorage.getItem('mongodb-docs') || '{}');
      if (docsObj) {
        return docsObj[key];
      }
    }
    return undefined;
  } catch {
    console.error('Error getting localStorage value');
    return undefined;
  }
};

export const setSessionValue = (key: string, value: string) => {
  try {
    if (isValidStorage) {
      const prevState = JSON.parse(window.sessionStorage.getItem('mongodb-docs') || '{}');
      sessionStorage.setItem('mongodb-docs', JSON.stringify({ ...prevState, [key]: value }));
    }
  } catch {
    console.error('Error setting session value');
  }
};

export const getSessionValue = (key: string) => {
  try {
    if (isValidStorage && JSON.parse(window.sessionStorage.getItem('mongodb-docs') || '{}')) {
      const docsObj = JSON.parse(window.sessionStorage.getItem('mongodb-docs') || '{}');
      if (docsObj) {
        return docsObj[key];
      }
    }
    return undefined;
  } catch {
    console.error('Error getting session value');
    return undefined;
  }
};
