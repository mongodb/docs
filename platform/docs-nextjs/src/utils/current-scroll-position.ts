import { isBrowser } from './is-browser';

// Function that returns current scroll position for analytics
export const currentScrollPosition = (): number => {
  if (isBrowser) {
    const percentage = Math.round((window.scrollY / (document.body.scrollHeight - window.innerHeight)) * 100);
    return Math.round(percentage / 10) * 10;
  }
  return 0;
};
