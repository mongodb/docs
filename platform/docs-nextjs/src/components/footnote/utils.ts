import type { ASTNode } from '@/types/ast';
import type { FootnoteNode, FootnoteReferenceNode } from '@/types/ast';
import { findAllKeyValuePairs } from '@/utils/find-all-key-value-pairs';
import type { FootnoteType } from './footnote-context';

export const getFootnotes = (nodes: ASTNode[]) => {
  const footnotes = findAllKeyValuePairs(nodes, 'type', 'footnote') as FootnoteNode[];
  const footnoteReferences = findAllKeyValuePairs(nodes, 'type', 'footnote_reference') as FootnoteReferenceNode[];
  const numAnonRefs = footnoteReferences.filter(
    (node) => !Object.prototype.hasOwnProperty.call(node, 'refname'),
  ).length;
  // We label our footnotes by their index, regardless of their names to
  // circumvent cases such as [[1], [#], [2], ...]
  return footnotes.reduce<Record<string, FootnoteType>>((map, footnote, index) => {
    if (footnote.name) {
      // Find references associated with a named footnote

      map[footnote.name] = {
        label: index + 1,
        references: getNamedFootnoteReferences(footnoteReferences, footnote.name),
      };
    } else {
      // Find references associated with an anonymous footnote
      // Replace potentially broken anonymous footnote ids
      footnote.id = `${index + 1}`;

      map[footnote.id] = {
        label: index + 1,
        references: getAnonymousFootnoteReferences(index, numAnonRefs),
      };
    }
    return map;
  }, {});
};

// Find all footnote_reference node IDs associated with a given footnote by
// that footnote's refname
const getNamedFootnoteReferences = (footnoteReferences: FootnoteReferenceNode[], refname: string) => {
  return footnoteReferences.filter((node) => node.refname === refname).map((node) => node.id);
};

// They are used infrequently, but here we match an anonymous footnote to its reference.
// The nth footnote on a page is associated with the nth reference on the page. Since
// anon footnotes and footnote references are anonymous, we assume a 1:1 pairing, and
// have no need to query nodes. If there are more anonymous footnotes than references,
// we may return an empty array
const getAnonymousFootnoteReferences = (index: number, numAnonRefs: number) => {
  return index > numAnonRefs ? [] : [`id${index + 1}`];
};
