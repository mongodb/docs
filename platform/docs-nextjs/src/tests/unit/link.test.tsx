import React from 'react';
import { render } from '@testing-library/react';
import Link from '@/components/link';
import { TestWrapper } from '../utils/test-wrapper';

const setup = ({ text, ...rest }: { text: string;[key: string]: unknown }) => {
  return render(
    <TestWrapper>
      <Link {...rest}>{text}</Link>
    </TestWrapper>
  );
};


describe('Link component renders a variety of strings correctly', () => {
  it('empty string', () => {
    const tree = setup({ to: '', text: 'Empty string' });
    expect(tree.asFragment()).toMatchSnapshot();
  });

  it('external URL', () => {
    const tree = setup({ to: 'http://mongodb.com', text: 'MongoDB Company' });
    expect(tree.asFragment()).toMatchSnapshot();
  });

  it('external MDB docs URL', () => {
    const tree = setup({ to: 'https://www.mongodb.com/docs/atlas/', text: 'MongoDB Atlas Docs' });
    expect(tree.asFragment()).toMatchSnapshot();
  });

  it('external MDB non-docs URL', () => {
    const tree = setup({ to: 'https://account.mongodb.com/account/', text: 'MongoDB Atlas Registration Page' });
    expect(tree.asFragment()).toMatchSnapshot();
  });

  it('external non-MDB docs URL', () => {
    const tree = setup({ to: 'https://safety.google/authentication/', text: 'Google 2-Step Verification' });
    expect(tree.asFragment()).toMatchSnapshot();
  });

  it('external URL with hidden external icon', () => {
    const tree = setup({ to: 'http://mongodb.com', hideExternalIcon: true, text: 'MongoDB Company' });
    expect(tree.asFragment()).toMatchSnapshot();
  });

  it('internal link', () => {
    const tree = setup({ to: 'drivers/c', text: 'C Driver', className: 'test-class' });
    expect(tree.asFragment()).toMatchSnapshot();
  });

  it('internal link with hash', () => {
    const tree = setup({ to: 'drivers/pymongo#installation', text: 'C Driver', className: 'test-class' });
    expect(tree.asFragment()).toMatchSnapshot();
  });

  it('internal link query param', () => {
    const tree = setup({ to: 'drivers/ruby?site=drivers', text: 'C Driver', className: 'test-class' });
    expect(tree.asFragment()).toMatchSnapshot();
  });

  it('internal link that already includes trailing slash', () => {
    const tree = setup({ to: 'drivers/nodejs/#installation', text: 'C Driver', className: 'test-class' });
    expect(tree.asFragment()).toMatchSnapshot();
  });

  it('identfies mailto links as external urls', () => {
    const tree = setup({ to: 'mailto:docs@mongodb.com', text: 'docs@mongodb.com' });
    expect(tree.asFragment()).toMatchSnapshot();
  });
});
