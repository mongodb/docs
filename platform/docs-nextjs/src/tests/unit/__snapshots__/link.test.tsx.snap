// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component renders a variety of strings correctly empty string 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-decoration-color: transparent;
  cursor: pointer;
  font-size: inherit;
  line-height: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: none;
  border: none;
  padding: 0;
  font-size: 13px;
  line-height: 20px;
  color: #016BF8;
  font-weight: 400;
  display: inline;
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

.emotion-0:hover,
.emotion-0:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-transition: text-decoration 150ms ease-in-out;
  transition: text-decoration 150ms ease-in-out;
  text-underline-offset: 4px;
  text-decoration-thickness: 2px;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:hover {
  text-decoration-color: #E8EDEB;
}

.emotion-0:focus {
  text-decoration-color: #016BF8;
}

.emotion-0>span>code,
.emotion-0>code {
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

<a
    class="lg-ui-0001 emotion-0"
    href=""
  >
    <span>
      Empty string
    </span>
  </a>
</DocumentFragment>
`;

exports[`Link component renders a variety of strings correctly external MDB docs URL 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-decoration-color: transparent;
  cursor: pointer;
  font-size: inherit;
  line-height: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: none;
  border: none;
  padding: 0;
  font-size: 13px;
  line-height: 20px;
  color: #016BF8;
  font-weight: 400;
  display: inline;
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

.emotion-0:hover,
.emotion-0:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-transition: text-decoration 150ms ease-in-out;
  transition: text-decoration 150ms ease-in-out;
  text-underline-offset: 4px;
  text-decoration-thickness: 2px;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:hover {
  text-decoration-color: #E8EDEB;
}

.emotion-0:focus {
  text-decoration-color: #016BF8;
}

.emotion-0>span>code,
.emotion-0>code {
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

<a
    class="lg-ui-0001 emotion-0"
    href="https://www.mongodb.com/docs/atlas/"
    target="_self"
  >
    <span>
      MongoDB Atlas Docs
    </span>
  </a>
</DocumentFragment>
`;

exports[`Link component renders a variety of strings correctly external MDB non-docs URL 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-decoration-color: transparent;
  cursor: pointer;
  font-size: inherit;
  line-height: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: none;
  border: none;
  padding: 0;
  font-size: 13px;
  line-height: 20px;
  color: #016BF8;
  font-weight: 400;
  display: inline;
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

.emotion-0:hover,
.emotion-0:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-transition: text-decoration 150ms ease-in-out;
  transition: text-decoration 150ms ease-in-out;
  text-underline-offset: 4px;
  text-decoration-thickness: 2px;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:hover {
  text-decoration-color: #E8EDEB;
}

.emotion-0:focus {
  text-decoration-color: #016BF8;
}

.emotion-0>span>code,
.emotion-0>code {
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

<a
    class="lg-ui-0001 emotion-0"
    href="https://account.mongodb.com/account/"
    target="_self"
  >
    <span>
      MongoDB Atlas Registration Page
    </span>
  </a>
</DocumentFragment>
`;

exports[`Link component renders a variety of strings correctly external URL 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-decoration-color: transparent;
  cursor: pointer;
  font-size: inherit;
  line-height: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: none;
  border: none;
  padding: 0;
  font-size: 13px;
  line-height: 20px;
  color: #016BF8;
  font-weight: 400;
  display: inline;
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

.emotion-0:hover,
.emotion-0:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-transition: text-decoration 150ms ease-in-out;
  transition: text-decoration 150ms ease-in-out;
  text-underline-offset: 4px;
  text-decoration-thickness: 2px;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:hover {
  text-decoration-color: #E8EDEB;
}

.emotion-0:focus {
  text-decoration-color: #016BF8;
}

.emotion-0>span>code,
.emotion-0>code {
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

<a
    class="lg-ui-0001 emotion-0"
    href="http://mongodb.com"
    target="_self"
  >
    <span>
      MongoDB Company
    </span>
  </a>
</DocumentFragment>
`;

exports[`Link component renders a variety of strings correctly external URL with hidden external icon 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-decoration-color: transparent;
  cursor: pointer;
  font-size: inherit;
  line-height: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: none;
  border: none;
  padding: 0;
  font-size: 13px;
  line-height: 20px;
  color: #016BF8;
  font-weight: 400;
  display: inline;
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

.emotion-0:hover,
.emotion-0:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-transition: text-decoration 150ms ease-in-out;
  transition: text-decoration 150ms ease-in-out;
  text-underline-offset: 4px;
  text-decoration-thickness: 2px;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:hover {
  text-decoration-color: #E8EDEB;
}

.emotion-0:focus {
  text-decoration-color: #016BF8;
}

.emotion-0>span>code,
.emotion-0>code {
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

<a
    class="lg-ui-0001 emotion-0"
    href="http://mongodb.com"
    target="_self"
  >
    <span>
      MongoDB Company
    </span>
  </a>
</DocumentFragment>
`;

exports[`Link component renders a variety of strings correctly external non-MDB docs URL 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-decoration-color: transparent;
  cursor: pointer;
  font-size: inherit;
  line-height: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: none;
  border: none;
  padding: 0;
  font-size: 13px;
  line-height: 20px;
  color: #016BF8;
  font-weight: 400;
  display: inline;
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

.emotion-0:hover,
.emotion-0:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-transition: text-decoration 150ms ease-in-out;
  transition: text-decoration 150ms ease-in-out;
  text-underline-offset: 4px;
  text-decoration-thickness: 2px;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:hover {
  text-decoration-color: #E8EDEB;
}

.emotion-0:focus {
  text-decoration-color: #016BF8;
}

.emotion-0>span>code,
.emotion-0>code {
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

.emotion-1 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: relative;
  bottom: 4px;
  left: -1px;
  height: 12px;
}

<a
    class="lg-ui-0001 emotion-0"
    href="https://safety.google/authentication/"
    rel="noopener noreferrer"
  >
    <span>
      Google 2-Step Verification
    </span>
    <svg
      alt=""
      aria-hidden="true"
      class="emotion-1"
      height="16"
      role="presentation"
      viewBox="0 0 16 16"
      width="16"
    >
      <path
        d="M13.823 2.4491C13.8201 2.30008 13.6999 2.17994 13.5509 2.17704L9.5062 2.09836C9.25654 2.09351 9.12821 2.39519 9.30482 2.5718L10.3856 3.65257L7.93433 6.10383C7.87964 6.15852 7.83047 6.21665 7.78683 6.27752L5.99909 8.06525C5.46457 8.59977 5.46457 9.4664 5.99909 10.0009C6.53361 10.5354 7.40023 10.5354 7.93475 10.0009L9.72249 8.21317C9.78336 8.16953 9.84148 8.12037 9.89618 8.06567L12.3474 5.61441L13.4282 6.69518C13.6048 6.87179 13.9065 6.74347 13.9016 6.4938L13.823 2.4491Z"
        fill="currentColor"
      />
      <path
        d="M7.25 3.12893C7.66421 3.12893 8 3.46472 8 3.87893C8 4.29315 7.66421 4.62893 7.25 4.62893H4C3.72386 4.62893 3.5 4.85279 3.5 5.12893V11.9929C3.5 12.2691 3.72386 12.4929 4 12.4929H10.864C11.1401 12.4929 11.364 12.2691 11.364 11.9929V8.75C11.364 8.33579 11.6998 8 12.114 8C12.5282 8 12.864 8.33579 12.864 8.75V11.9929C12.864 13.0975 11.9686 13.9929 10.864 13.9929H4C2.89543 13.9929 2 13.0975 2 11.9929V5.12893C2 4.02436 2.89543 3.12893 4 3.12893H7.25Z"
        fill="currentColor"
      />
    </svg>
  </a>
</DocumentFragment>
`;

exports[`Link component renders a variety of strings correctly identfies mailto links as external urls 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-decoration-color: transparent;
  cursor: pointer;
  font-size: inherit;
  line-height: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: none;
  border: none;
  padding: 0;
  font-size: 13px;
  line-height: 20px;
  color: #016BF8;
  font-weight: 400;
  display: inline;
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

.emotion-0:hover,
.emotion-0:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-transition: text-decoration 150ms ease-in-out;
  transition: text-decoration 150ms ease-in-out;
  text-underline-offset: 4px;
  text-decoration-thickness: 2px;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:hover {
  text-decoration-color: #E8EDEB;
}

.emotion-0:focus {
  text-decoration-color: #016BF8;
}

.emotion-0>span>code,
.emotion-0>code {
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

<a
    class="lg-ui-0001 emotion-0"
    href="mailto:docs@mongodb.com"
    target="_self"
  >
    <span>
      docs@mongodb.com
    </span>
  </a>
</DocumentFragment>
`;

exports[`Link component renders a variety of strings correctly internal link 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-decoration-color: transparent;
  line-height: 13px;
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

.emotion-0>span>code,
.emotion-0>code {
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

.emotion-0:focus,
.emotion-0:hover {
  text-decoration-line: underline;
  -webkit-transition: text-decoration 150ms ease-in-out;
  transition: text-decoration 150ms ease-in-out;
  text-underline-offset: 4px;
  text-decoration-thickness: 2px;
}

.emotion-0:focus {
  text-decoration-color: #016BF8;
  outline: none;
}

.emotion-0:hover {
  text-decoration-color: #E8EDEB;
}

<a
    class="emotion-0 test-class"
    href="/drivers/c/"
  >
    C Driver
  </a>
</DocumentFragment>
`;

exports[`Link component renders a variety of strings correctly internal link query param 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-decoration-color: transparent;
  line-height: 13px;
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

.emotion-0>span>code,
.emotion-0>code {
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

.emotion-0:focus,
.emotion-0:hover {
  text-decoration-line: underline;
  -webkit-transition: text-decoration 150ms ease-in-out;
  transition: text-decoration 150ms ease-in-out;
  text-underline-offset: 4px;
  text-decoration-thickness: 2px;
}

.emotion-0:focus {
  text-decoration-color: #016BF8;
  outline: none;
}

.emotion-0:hover {
  text-decoration-color: #E8EDEB;
}

<a
    class="emotion-0 test-class"
    href="/drivers/ruby/?site=drivers"
  >
    C Driver
  </a>
</DocumentFragment>
`;

exports[`Link component renders a variety of strings correctly internal link that already includes trailing slash 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-decoration-color: transparent;
  line-height: 13px;
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

.emotion-0>span>code,
.emotion-0>code {
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

.emotion-0:focus,
.emotion-0:hover {
  text-decoration-line: underline;
  -webkit-transition: text-decoration 150ms ease-in-out;
  transition: text-decoration 150ms ease-in-out;
  text-underline-offset: 4px;
  text-decoration-thickness: 2px;
}

.emotion-0:focus {
  text-decoration-color: #016BF8;
  outline: none;
}

.emotion-0:hover {
  text-decoration-color: #E8EDEB;
}

<a
    class="emotion-0 test-class"
    href="/drivers/nodejs/#installation"
  >
    C Driver
  </a>
</DocumentFragment>
`;

exports[`Link component renders a variety of strings correctly internal link with hash 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-decoration-color: transparent;
  line-height: 13px;
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

.emotion-0>span>code,
.emotion-0>code {
  color: var(--link-color-primary);
  font-weight: var(--link-font-weight);
}

.emotion-0:focus,
.emotion-0:hover {
  text-decoration-line: underline;
  -webkit-transition: text-decoration 150ms ease-in-out;
  transition: text-decoration 150ms ease-in-out;
  text-underline-offset: 4px;
  text-decoration-thickness: 2px;
}

.emotion-0:focus {
  text-decoration-color: #016BF8;
  outline: none;
}

.emotion-0:hover {
  text-decoration-color: #E8EDEB;
}

<a
    class="emotion-0 test-class"
    href="/drivers/pymongo/#installation"
  >
    C Driver
  </a>
</DocumentFragment>
`;
